spec/03-plugins/25-response-rate-limiting/04-access_spec.lua

```diff
@@ -55,834 +51,776 @@ local function flush_redis()
-      local bp
-      local dao
-      setup(function()
-        helpers.kill_all()
-        flush_redis()
-        local _
-        bp, _, dao = helpers.get_db_utils(strategy, {
-          "routes",
-          "services",
-          "plugins",
-          "consumers",
-          "keyauth_credentials",
-        })
-        dao.db:truncate_table("response_ratelimiting_metrics")
-
-        local consumer = bp.consumers:insert {
-          custom_id = "provider_125",
-        }
+    assert(helpers.start_kong({
+      database   = strategy,
+      nginx_conf = "spec/fixtures/custom_nginx.template",
+    }))
+  end)
+
+  teardown(function()
+    helpers.stop_kong(nil, true)
+  end)
+
+  it("blocks when the consumer exceeds their quota, no matter what service/route used", function()
+    test_limit("/response-headers?apikey=apikey126&x-kong-limit=video=1", "test%d.com")
+  end)
+end)
+
+describe(fmt("#flaky Plugin: response-ratelimiting (access - global) with policy: #%s [#%s]", policy, strategy), function()
+
+  setup(function()
+    local bp = init_db(strategy, policy)
+
+    -- global plugin (not attached to route, service or consumer)
+    bp.response_ratelimiting_plugins:insert({
+      config = {
+        fault_tolerant = false,
+        policy         = policy,
+        redis_host     = REDIS_HOST,
+        redis_port     = REDIS_PORT,
+        redis_password = REDIS_PASSWORD,
+        redis_database = REDIS_DATABASE,
+        limits         = { video = { second = ITERATIONS } },
+      }
+    })
+
+    for i = 1, ITERATIONS do
+      bp.routes:insert({ hosts = { fmt("test%d.com", i) } })
+    end
 
-        bp.key_auth_plugins:insert()
+    assert(helpers.start_kong({
+      database   = strategy,
+      nginx_conf = "spec/fixtures/custom_nginx.template",
+    }))
+  end)
+
+  teardown(function()
+    helpers.stop_kong(nil, true)
+  end)
+
+  before_each(function()
+    wait()
+  end)
+
+  it("blocks if exceeding limit", function()
+    wait()
+    for i = 1, ITERATIONS do
+      local res = proxy_client():get("/response-headers?x-kong-limit=video=1", {
+        headers = { Host = fmt("test%d.com", i) },
+      })
+      assert.res_status(200, res)
+    end
 
-        bp.keyauth_credentials:insert {
-          key      = "apikey125",
-          consumer = { id = consumer.id },
+    ngx.sleep(SLEEP_TIME) -- Wait for async timer to increment the limit
+
+    -- last query, while limit is ITERATIONS/second
+    local res = proxy_client():get("/response-headers?x-kong-limit=video=1", {
+      headers = { Host = "test1.com" },
+    })
+    assert.res_status(429, res)
+    assert.equal(0, tonumber(res.headers["x-ratelimit-remaining-video-second"]))
+    assert.equal(ITERATIONS, tonumber(res.headers["x-ratelimit-limit-video-second"]))
+  end)
+end)
+
+describe(fmt("#flaky Plugin: response-ratelimiting (fault tolerance) with policy: #%s [#%s]", policy, strategy), function()
+--  if policy == "cluster" then
```

Comment:
What is all this commented-out code? Is it needed?

Reply:
yes, it should be fixed eventually